### ISSUE TITLE
Can't use 'bridge' default network after docker service upgrade

### ISSUE THREAD

**Issue by icez** [docker/docker#27280]
Hello,

today we're just upgraded from docker-engine-1.11.2-1.el7.centos.x86_64 to docker-engine-1.12.1-1.el7.centos.x86_64

right after upgrade was done, the container using default 'bridge' network cannot start with the following error:

Error response from daemon: failed to create endpoint <<ctname>> on network bridge: network d82988963ee05e9d0ea8401571f52559a7507babc32c8cc2d422ce1ac6d8eb44 does not exist

The workaround is by creating a new network (called 'newbridge') and link the ct to this bridge instead.

After 'docker network ls' I also found that there's no 'host' and 'null' network.
The network ID seems to match on the list of 'docker network ls' but it can't be used

Server's running on physical server.

[root@docker1 log]# docker network ls --no-trunc
NETWORK ID                                                         NAME                                DRIVER              SCOPE
d82988963ee05e9d0ea8401571f52559a7507babc32c8cc2d422ce1ac6d8eb44   bridge                              bridge              local
7ad8ff0c449eb6894feddc746889b7ab73ba104697d061b452a671cc32149d30   newbridge                           bridge              local

[root@docker1 docker]# docker network inspect d82988963ee0
[
    {
        "Name": "bridge",
        "Id": "d82988963ee05e9d0ea8401571f52559a7507babc32c8cc2d422ce1ac6d8eb44",
        "Scope": "local",
        "Driver": "bridge",
        "EnableIPv6": false,
        "IPAM": {
            "Driver": "default",
            "Options": null,
            "Config": [
                {
                    "Subnet": "192.168.0.0/20",
                    "Gateway": "192.168.0.1"
                }
            ]
        },
        "Internal": false,
        "Containers": {},
        "Options": {
            "com.docker.network.bridge.default_bridge": "true",
            "com.docker.network.bridge.enable_icc": "true",
            "com.docker.network.bridge.enable_ip_masquerade": "true",
            "com.docker.network.bridge.host_binding_ipv4": "0.0.0.0",
            "com.docker.network.bridge.name": "docker0",
            "com.docker.network.driver.mtu": "1500"
        },
        "Labels": {}
    }
]

[root@docker1 log]# docker version
Client:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.1
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   23cf638
 Built:
 OS/Arch:      linux/amd64

[root@docker1 log]# docker info
Containers: 82
 Running: 40
 Paused: 0
 Stopped: 42
Images: 867
Server Version: 1.12.1
Storage Driver: devicemapper
 Pool Name: docker-8:2-38274796-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: ext4
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 42.85 GB
 Data Space Total: 107.4 GB
 Data Space Available: 64.52 GB
 Metadata Space Used: 57.56 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.09 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2016-06-09)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: bridge overlay
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-327.36.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 15.47 GiB
Name: docker1.********
ID: CFHX:W4WY:DQ4V:S6FR:LEJL:GLHE:ZNZC:NMGG:BTTQ:TPJ6:RJSD:CON3
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): true
 File Descriptors: 265
 Goroutines: 356
 System Time: 2016-10-11T15:55:37.803874449+07:00
 EventsListeners: 0
Username: ***************
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8

**Comment by mrjana** [comment]
ping @aboch

**Comment by aboch** [comment]
@icez Please attach your `/var/lib/docker/network/files/local-kv.db` file.
It should allow me to reproduce the issue and checkwhat is wrong.
Thanks

**Comment by icez** [comment]
@aboch here it is

[local-kv.zip](https://github.com/docker/docker/files/523294/local-kv.zip)

**Comment by aboch** [comment]
Thanks @icez .
Unfortunately I was not able to recreate the issue using your file (it has probably changed after you started docker 1.12.1, it in fact contains the `newbridge` network), or from a clean state for which I am reporting the logs below:

```
$ # starting 1.11.2 with no local store file
$
$ docker version | grep Vers
 Version:      1.11.2
 Version:      1.11.2
$
$ docker run -d --name c0 --net bridge busybox top
407e7cc8c1f0cc13231e777a4a303740361a08a427382208c62d07d287461853
$
$ docker network ls
NETWORK ID          NAME                DRIVER
b5870d65f1bc        bridge              bridge              
99d97950811c        host                host                
05edb8eee2d4        none                null                
$ 
$ # stopping 1.11.2 and starting 1.12.2
$ 
$ docker version | grep Ver
 Version:      1.12.2
 Version:      1.12.2
$
$ docker network ls
NETWORK ID          NAME                DRIVER              SCOPE
b55cc526352d        bridge              bridge              local               
6050cba8ea31        host                host                local               
369ae3bd9ed6        none                null                local               
$
$  docker start c0
c0
$ 
```

Also I did not hit the missing `host` and `null` networks issue.

Note: I tested the 1.11.2 -> 1.12.2 upgrade (by CTRL+C current daemon, replacing the docker binaries and starting), instead of upgrading to 1.12.1. It should be equivalent, I do not remember any change in this area between 12.1 and 12.2.

How did you start your containers on `bridge` network on 1.11.2, anything special ?
What options are you passing at docker daemon start ?
If you have time to try the test I did, go back to 1.11.2 binaries and then move to the 1.12.2, can you reproduce the issue ?

**Comment by aboch** [comment]
Also, in the docker logs, do you see any failures warning/errors. 
Maybe the update process went through few restart cycles, because of some other bug.

**Comment by icez** [comment]
Sorry for late reply. I've tried reproduce the issue by using the file local-kv on other machine but can't reproduce the error either.

@aboch I don't see any error in log file at the time. Tried enable debug to see but there's no error on the log (except that the container cannot start because network does not working properly.)

Also, I've rename /var/lib/docker to /var/lib/docker.bak and re-create empty folder in place. Then start the docker service again. This time docker network work fine.

In my opinion, it seems like the docker somehow cannot load the network plugin properly so it lead to the network drivers not able to work. But I still don't understand why that I'm able to create new brirdge  (which should use the same bridge driver) and use that for the CT network instead.
